### PR TITLE
Feature/red 443 blocking grades

### PR DESCRIPTION
**Pull request by PanopticaRising**

### CODE REVIEW

File: src/features/courses/course-controller.ts [hunk 1]
```diff
@@ -2857,33 +2871,121 @@ class CourseController {
...
+                    ]
+                }
+            ]
+        });
+
+        let attributes;
+        let gradeInstanceAttributes;
+        let workbookAttributes;
+        if (userRole === Role.STUDENT && (topicAssessmentInfo?.showTotalGradeImmediately === false || topicAssessmentInfo?.showItemizedResults === false)) {
+            logger.warn('Setting attributes to empty');
```
Comment (PanopticaRising) [inline src/features/courses/course-controller.ts:2918]: Gotta remove this.

File: src/features/courses/course-controller.ts [hunk 2]
```diff
@@ -2857,33 +2871,130 @@ class CourseController {
...
+                        }
+                    ]
+                }
+            ]
+        });
+
+        let attributes;
+        let gradeInstanceAttributes;
+        let workbookAttributes;
+        if (userRole === Role.STUDENT && (topicAssessmentInfo?.showTotalGradeImmediately === false || topicAssessmentInfo?.showItemizedResults === false)) {
```
Comment (red-tt-bear) [inline src/features/courses/course-controller.ts:2917]: Shouldn't this be

userRole is student AND
showTotalGradeImmediately AND
showItemizedResults

The or with the falses seems off

Comment (red-tt-bear) [reply src/features/courses/course-controller.ts:2917]: or maybe it should just be that they are true? unless there is a double negative somewhere?

Comment (red-tt-bear) [reply src/features/courses/course-controller.ts:2917]: Oh unless you're using this to filter the attributes and relying on them being undefined if not entered here to show everything?

Without types on the variables it's pretty confusing

**Review by red-tt-bear — APPROVED** [review]

**Pull request merged**